GVFS/GVFS.Platform.Windows/WindowsFileSystemVirtualizer.cs

```diff
@@ -1005,29 +1005,16 @@ private HResult QueryFileNameHandler(string virtualPath)
                         GitCommandLineParser gitCommand = new GitCommandLineParser(this.Context.Repository.GVFSLock.GetLockedGitCommand());
                         if (gitCommand.IsValidGitCommand)
                         {
-                            string directoryPath = Path.Combine(this.Context.Enlistment.WorkingDirectoryRoot, virtualPath);
-                            HResult hr = this.virtualizationInstance.MarkDirectoryAsPlaceholder(
-                                directoryPath,
-                                FolderContentId,
-                                PlaceholderVersionId);
-
-                            if (hr == HResult.Ok)
-                            {
-                                this.FileSystemCallbacks.OnPlaceholderFolderCreated(virtualPath, triggeringProcessImageFileName);
-                            }
-                            else
-                            {
-                                EventMetadata metadata = this.CreateEventMetadata(virtualPath);
-                                metadata.Add("isDirectory", isDirectory);
-                                metadata.Add("triggeringProcessId", triggeringProcessId);
-                                metadata.Add("triggeringProcessImageFileName", triggeringProcessImageFileName);
-                                metadata.Add("HResult", hr.ToString());
-                                this.Context.Tracer.RelatedError(metadata, nameof(this.NotifyNewFileCreatedHandler) + "_" + nameof(this.virtualizationInstance.MarkDirectoryAsPlaceholder) + " error");
-                            }
+                            this.MarkDirectoryAsPlaceholder(virtualPath, isDirectory, triggeringProcessId, triggeringProcessImageFileName);

```

Comment:
What is the expected behavior if git creates a directory that's currently excluded?

Reply:
The directory will be marked as a placeholder and start projecting and it should get removed once a index change happens if nothing adds to the modified paths and they are all placeholders still but if the skipworktree bit is turned off, paths will get added to the modified paths and the folder will not get removed.  I did some testing around the reset scenarios and this is what happened.